### ISSUE TITLE
Feature request: Detect (and warn) if in conflict with debugger

### ISSUE THREAD

**Issue by posita** [coveragepy/coveragepy#1337]
This came up in https://github.com/microsoft/debugpy/issues/863#issuecomment-1059171271, where `debugpy` silently fails to hit API-set breakpoints when using coverage (e.g., `python -m debugpy … -m pytest --cov …`). Is this conflict something that can be detected at runtime as @fabioz suggests in that comment?

I'm aware of #647. This is a different request. I'm not asking whether coverage can be made to work with debuggers, only to detect whether it was overriding prior registrations and signal that to the user so they'd have some insight into why their debuggers weren't working as expected.

**Comment by nedbat** [comment]
Yes, I suppose coverage could warn that a trace function is in place when it starts.
